holocron/trainer/core.py

```diff
@@ -364,6 +364,52 @@ def _eval_metrics_str(eval_metrics: Dict[str, float]) -> str:
                 f"(Acc@1: {eval_metrics['acc1']:.2%}, Acc@5: {eval_metrics['acc5']:.2%})")
 
 
+class BinaryClassificationTrainer(Trainer):
+    """Image binary classification trainer class
+
+    Args:
+        model (torch.nn.Module): model to train
+        train_loader (torch.utils.data.DataLoader): training loader
+        val_loader (torch.utils.data.DataLoader): validation loader
+        criterion (torch.nn.Module): loss criterion
+        optimizer (torch.optim.Optimizer): parameter optimizer
+        gpu (int, optional): index of the GPU to use
+        output_file (str, optional): path where checkpoints will be saved
+    """
+
+    @torch.no_grad()
+    def evaluate(self) -> Dict[str, float]:
+        """Evaluate the model on the validation set
+
+        Returns:
+            dict: evaluation metrics
+        """
+
+        self.model.eval()
+
+        val_loss, top1, num_samples = 0., 0, 0
+        for x, target in self.val_loader:
+            x, target = self.to_cuda(x, target)
+
+            # Forward
+            out = self.model(x)
+            # Loss computation
+            val_loss += self.criterion(out, target).item()
+
+            top1 += torch.sum(abs(target - torch.sigmoid(out)) < 0.5).item()
```

Comment:
Mmmh, I see two things I'm not sure how to handle:
- in the other trainers, you don't assume how the predictions are made (whether it's a sigmoid or something else). I'm not sure how we could achieve it here (perhaps a use_sigmoid flag set in the constructor?)
- so the `abs(target - torch.sigmoid(out)) < 0.5` is actually dangerous, because this assumes that the target is binary. While the source target  is, if the user add something like label smoothing you can end up with a target of 0.9. And with your criteria, if the predicted probability was 0.41, you would wrongly say that the prediction was correct. I would suggest `(target >=0.5) == (torch.sigmoid(out) >= 0.5)` which would solve the issue.

Reply:
In both cases I was a little too focused on my use case. But yes the flag in the constructor seems to me insdispensable and I will add the modification for the second point. Thanks !

Reply:
ok I get it know, is about the size of the output tensor :/